flow/scripts/util.tcl

```diff
@@ -3,7 +3,7 @@ proc log_cmd {cmd args} {
   set log_cmd "$cmd[join [lmap arg $args {format " %s" [expr {[string match {* *} $arg] ? "\"$arg\"" : "$arg"}]}] ""]"
   puts $log_cmd
   set start [clock seconds]
-  $cmd {*}$args
+  uplevel 1 [list $cmd {*}$args]
```

Comment:
Nice! I had to read line 6 twice before I realized it was a concatenation, but either is idiomatic tcl:

    uplevel 1 [concat [list $cmd] $args]